components/lists/SpeakersList.tsx

```diff
@@ -1,17 +1,73 @@
+import { useTheme } from '@react-navigation/native';
+import { useRouter } from 'expo-router';
 import React from 'react';
-import { View } from 'react-native';
-
-// TODO:- Implement a list that displays speaker images and names
-/**
- * - limit the speakers to 15
- * - add a link to [speaker] page when the image is pressed
- * - handle text if the speaker name is long and center it
- * - each image should have a border as seen in the designs
- * - write tests for the component
- */
+import type { ListRenderItemInfo } from 'react-native';
+import { FlatList, Image, StyleSheet, View } from 'react-native';
+import type { Speaker } from '../../global';
+import { Speakers } from '../../mock/speakers';
+import ViewAllButton from '../buttons/ViewAllButton';
+import Row from '../common/Row';
+import Space from '../common/Space';
+import StyledText from '../common/StyledText';
 
 const SpeakersList = () => {
-  return <View />;
+  const { colors } = useTheme();
+  const router = useRouter();
+  const speakers = Speakers.data.slice(0, 15); // filter speakers to 15
+  return (
+    <View style={styles.list}>
+      <Row>
+        <StyledText font="bold" size="lg" style={{ color: colors.primary }}>
+          Speakers
+        </StyledText>
+        <ViewAllButton onPress={() => router.push('/speakers')} label={Speakers.data.length.toString()} />
+      </Row>
+      <Space size={16} />
+      <FlatList
+        data={speakers}
+        renderItem={({ item }: ListRenderItemInfo<Speaker>) => (
+          <View style={styles.item}>
+            <Image
+              source={{ uri: item.avatar }}
+              style={[styles.image, { borderColor: colors.tint }]}
+              resizeMode="contain"
+            />
+            <StyledText size="sm" font="medium" style={styles.description} numberOfLines={2}>
+              {item.name}
+            </StyledText>
+          </View>
```

Comment:
So I have noticed dummy data `sessions.ts` does not have start_time and room. ( Sucks, ik)
Last year I noted the same and I ended up using data from `schedule.ts`
The sessions on *homescreen* would be rendered based on the day eg if it's day 1, it shows day 1 sessions and so forth.
Also the same for speakers, but the problem with that is some days didn't have specific speakers so this is optional.

Uhm this might be a huge change but could you refactor this?

Reply:
its okay, for now I just added a simple function that gets the room and start time from the schedule

`
const getSessionTimeAndLocation = (slug: string) => {
    for (const key in Schedule.data) {
      const sessionData = Schedule.data[key];
      const session = sessionData?.find((item: SessionForSchedule) => item.slug === slug);
      if (session) {
        const startTime = session.start_time.split(':').slice(0, 2).join(':');
        return `@ ${startTime}  |  Room ${session?.rooms[0]?.title}`;
      }
    }
    return '';
  };
`

Reply:
> its okay, for now I just added a simple function that gets the room and start time from the schedule
> 
> `const getSessionTimeAndLocation = (slug: string) => { for (const key in Schedule.data) { const sessionData = Schedule.data[key]; const session = sessionData?.find((item: SessionForSchedule) => item.slug === slug); if (session) { const startTime = session.start_time.split(':').slice(0, 2).join(':'); return`@ ${startTime} | Room ${session?.rooms[0]?.title}`; } } return ''; }; `

Ok, could you push this code I check it out